core/src/main/java/org/apache/iceberg/rest/requests/ReportMetricsRequest.java

```diff
@@ -54,16 +55,20 @@ default void validate() {
   }
 
   static ReportMetricsRequest of(MetricsReport report) {
-    ReportType reportType = null;
+    ReportType reportType = ReportType.UNKNOWN;
     if (report instanceof ScanReport) {
       reportType = ReportType.SCAN_REPORT;
     } else if (report instanceof CommitReport) {
       reportType = ReportType.COMMIT_REPORT;
     }
 
-    Preconditions.checkArgument(
-        null != reportType, "Unsupported report type: %s", report.getClass().getName());
-
     return ImmutableReportMetricsRequest.builder().reportType(reportType).report(report).build();
```

Comment:
we don't fallback to returning `ReportMetricsRequest unknown()` because should prefer the `report` that was passed in to this method in L57

Reply:
Is this for cases where an unknown metrics report is passed by a user rather than by Iceberg? I guess that makes sense, but I wouldn't expect it to happen.

Reply:
I wouldn't expect it to happen either, but I think it's more correct to pass-through whatever the `report` was in the first place (rather than defaulting to an unknown impl).